evm/broadcast-transactions.js

```diff
@@ -0,0 +1,112 @@
+                    const response = await sendTx(transaction.signedTx, provider);
+
+                    if (response.error || !isValidJSON(response) || response.status !== 1) {
+                        const error = `Execution failed${
+                            response.status ? ` with txHash: ${response.transactionHash}` : ` with msg: ${response.message}`
+                        }`;
+                        throw new Error(error);
+                    }
+
+                    // Update the transaction status and store transaction hash
+                    transaction.status = 'SUCCESS';
+                    transaction.transactionHash = response.transactionHash;
+                    printInfo(`Transactions executed successfully ${response.transactionHash}`);
+                } catch (error) {
+                    // Update the transaction status and store error message
+                    transaction.status = 'FAILED';
+                    transaction.error = error.message;
+                    printError(`Transaction failed with error: ${error.message}`);
+                }
+            }
+        }
+
+        // Write back the updated JSON object to the file
+        signersData[signerAddress] = transactions;
+        const filePath = getFilePath(directoryPath, fileName);
+        fs.writeFileSync(filePath, JSON.stringify(signersData, null, 2));
+    } catch (error) {
+        printError('Error processing transactions:', error.message);
+    }
+}
+
+async function main(options) {
+    // TODO: Enable multiple scripts to use offlineSigning
+    const { directoryPath, fileName, rpcUrl, signerAddress } = options;
+    const provider = new JsonRpcProvider(rpcUrl);
+    const network = await provider.getNetwork();
+
+    if (!options.yes) {
+        const anwser = fs.readlineSync.question(
+            `Proceed with the broadcasting of all pending signed transactions for address ${chalk.green(
+                signerAddress,
+            )} on network ${chalk.green(network.name)} with chainId ${chalk.green(network.chainId)} ${chalk.green('(y/n)')} `,
+        );
+        if (anwser !== 'y') return;
+    }
+
+    await processTransactions(directoryPath, fileName, provider, signerAddress);
+}
+
+const program = new Command();
+
+program.name('broadcast-transactions').description('Broadcast all the pending signed transactions of the signer');
+
+program.addOption(
+    new Option('-d, --directoryPath <directoryPath>', 'The folder where all the signed tx files are stored').makeOptionMandatory(true),
+);
+program.addOption(new Option('-f, --fileName <fileName>', 'The fileName where the signed tx are stored').makeOptionMandatory(true));
+program.addOption(
+    new Option('-r, --rpcUrl <rpcUrl>', 'The rpc url for creating a provider to broadcast the transactions').makeOptionMandatory(true),
+);
+program.addOption(new Option('-s, --signerAddress <signerAddress>', 'private key').makeOptionMandatory(true));
```

Comment:
signer address shouldn't be needed for broadcasting since it's already in the tx and can be recovered from that if needed. Might even have different signers for the txs. I'd say just print the tx hashes for now (or if you want dump to a txs.json file with broadcast status) to keep it simpler.